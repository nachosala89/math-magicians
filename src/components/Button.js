import PropTypes from 'prop-types';
import React from 'react';

const Button = (props) => {
  const handleClickButton = () => {
    const { input, handleClickButton } = props;
    handleClickButton(input);
  };

  const { input, btnClass } = props;
  const classes = `calc-btn col-md-1 py-3 ${btnClass}`;
  return (
    <button className={classes} type="button" onClick={handleClickButton}>{input}</button>
  );
};

Button.propTypes = {
  input: PropTypes.string.isRequired,
  btnClass: PropTypes.string.isRequired,
  handleClickButton: PropTypes.func.isRequired,
};

export default Button;
