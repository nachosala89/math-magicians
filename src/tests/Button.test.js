import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

test('renders correctly', () => {
  const handleClickButton = () => {};
  const button = renderer
    .create(<Button btnClass="gray-btn w-25" input="AC" handleClickButton={handleClickButton} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
