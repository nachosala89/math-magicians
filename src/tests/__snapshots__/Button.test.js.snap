// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="calc-btn gray-btn w-25"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
